signer/src/emily_client.rs

```diff
@@ -126,6 +138,13 @@ impl EmilyInteract for EmilyClient {
             .collect()
     }
 
+    async fn set_chainstate(&self, chainstate: &Chainstate) -> Result<Chainstate, Error> {
+        chainstate_api::set_chainstate(&self.config, chainstate.clone())
```

Comment:
It makes perfect sense! I initially kept it as a reference to maintain consistency with the other signatures in `EmilyInteract`, but I can switch to taking ownership as you suggested.

One thing to note is that making this change will require adding a clone in the [ApiFallbackClient<EmilyClient> implementation](89b7625471cdc7c9d7e0113a504c2db48eed4097737d2671eab2c053fc629fe2R200), where the Chainstate needs to be passed to an Fn closure and the compiler complains if it's not cloned. However, this still makes sense since that function operates at a higher level, so the trade-off is reasonable. I'll change it tomorrow

Reply:
Ohh yeah. Well for now we can use the fact that we do not need to use the `exec` function for this call to Emily and we can just do this there:
```rust
self.get_client().set_chainstate(chainstate).await
```